### PR TITLE
Add missing trace decorator

### LINKED ISSUE #8: Add `traced/1` decorator
Our documentation is currently incorrect, as it claims that a `traced/1` decorator exists that would allow for adding context to the parent span.

### PR DESCRIPTION
**Pull request by zachdaniel**
Fixes #8

### CODE REVIEW

**Comment by zachdaniel** [comment]
@asummers @driv3r feel free to review :)

File: lib/trace_decorator.ex [hunk 1]
```diff
@@ -45,6 +45,38 @@ defmodule Spandex.TraceDecorator do
...
   end
 
+  def trace(attributes, body, context) do
+    quote do
+      if Spandex.disabled?() do
+        unquote(body)
+      else
+        attributes = Enum.into(unquote(attributes), %{})
+
+        name = attributes[:name] || "#{unquote(context.name)}/#{unquote(context.arity)}"
```
Comment (driv3r) [inline lib/trace_decorator.ex]: We could also extract name generation to separate method, i.e.

```elixir
# in decorator
name = context |> gen_name(attributes) |> unquote()

# later on
def gen_name(%{name: name, arity: arity}, name: nil), do: "#{name}/#{arity}"
def gen_name(_context, name: name), do: name
```

maybe same could be done for exception handler

File: lib/trace_decorator.ex [hunk 2]
```diff
@@ -45,6 +45,38 @@ defmodule Spandex.TraceDecorator do
...
+        _ =
+          case Spandex.start_trace(name) do
+            {:ok, trace_id} ->
+              _ = Spandex.update_span(attributes)
```
Comment (driv3r) [inline lib/trace_decorator.ex]: why we assign the result, if we don't care for it?

File: lib/trace_decorator.ex [hunk 3]
```diff
@@ -45,6 +45,38 @@ defmodule Spandex.TraceDecorator do
...
+              {:error, error}
+          end
+        try do
+          unquote(body)
+        rescue
+          exception ->
+            stacktrace = System.stacktrace
+          _ = Spandex.span_error(exception)
+
+          reraise(exception, stacktrace)
```
Comment (driv3r) [inline lib/trace_decorator.ex]: `_ = Spandex...` and `reraise` should have one more indentation

File: lib/trace_decorator.ex [hunk 4]
```diff
@@ -45,6 +45,38 @@ defmodule Spandex.TraceDecorator do
     end
   end
 
+  def trace(attributes, body, context) do
```
Comment (zachdaniel) [inline lib/trace_decorator.ex]: I agree, but this gave me issues when I tried it due to these being used in a macro somewhere. I'll try it again though :)

File: lib/trace_decorator.ex (hunk 1 above)
Comment (zachdaniel) [inline lib/trace_decorator.ex]: :+1:

File: lib/trace_decorator.ex (hunk 2 above)
Comment (zachdaniel) [inline lib/trace_decorator.ex]: Static analysis tools, like dialyzer, complain if there is a function that returns a type of {:ok | :error, term} if that type is not destructured and/or used by anything. `_ = operation` thus means "this operation can fail, and I'm acknowledging that I don't care".

**Comment by zachdaniel** [comment]
Need to add to the readme to allow Spandex logger metadata!

**Comment by driv3r** [comment]
:+1:

**Pull request merged**